refactor: cleanup index logging

Sjors opened the pull request:
This PR removes the use of `__func__` from index logging, since we have `-logsourcelocations`.

It also improves readability by putting `GetName()` in a more logical place.

Before

> coinstatsindex: best block of the index not found. Please rebuild the index.

After:

> best block of coinstatsindex not found. Please rebuild the index.

I found myself maintaining this commit as part of https://github.com/Sjors/bitcoin/pull/86, but since that might never land here, it seemed better to split it into its own PR (or get rid of it).

dergoegge commented:
While we're touching all of these, it might be worthwhile to change to `LogWarn/LogError` where appropriate.

dergoegge commented on src/index/blockfilterindex.cpp:
```diff
@@ -210,15 +210,15 @@ size_t BlockFilterIndex::WriteFilterToDisk(FlatFilePos& pos, const BlockFilter&
-            LogPrintf("%s: Failed to open filter file %d\n", __func__, pos.nFile);
+            LogInfo("Failed to open filter file %d", pos.nFile);
             return 0;
         }
         if (!last_file.Truncate(pos.nPos)) {
-            LogPrintf("%s: Failed to truncate filter file %d\n", __func__, pos.nFile);
+            LogInfo("Failed to truncate filter file %d", pos.nFile);
             return 0;
         }
         if (!last_file.Commit()) {
-            LogPrintf("%s: Failed to commit filter file %d\n", __func__, pos.nFile);
+            LogInfo("Failed to commit filter file %d", pos.nFile);
```
These can all be `LogError`

Sjors committed:
refactor: cleanup index logging

- don't log function name
- take into account that GetName() always ends with " index"
- replace deprecated LogPrintf with LogInfo
- remove trailing \n
- adjusted log level where needed

l0rinc commented:
Lightweight code review ACK c18bf0bd9be632e54591493a5309e9ed4020f2e5
